agent/connect/ca/provider_vault_test.go

```diff
@@ -51,6 +53,42 @@ func TestVaultCAProvider_SecondaryActiveIntermediate(t *testing.T) {
 	require.NoError(err)
 }
 
+func TestVaultCAProvider_RenewToken(t *testing.T) {
+	t.Parallel()
+	skipIfVaultNotPresent(t)
```

Comment:
@kyhavlov @alvin-huang The CodeCov [diff](https://codecov.io/gh/hashicorp/consul/compare/26d61c6b139cce1e555935ac3effb504171c6946...bb40e4c623e79e9f9681d931aa2daae2d46335a9/diff) is showing that most of the `NewVaultProvider` isn't being exercised by tests, which makes me think this is exiting early for some reason, any ideas?

Reply:
Ah, the test is running here: https://app.circleci.com/pipelines/github/hashicorp/consul/13261/workflows/dba860c7-f927-495b-91e3-1b55f3da889a/jobs/251660

They ran in a separate job (so that we don't have to setup the test environment for tests that don't need it). But this job is not uploading a coverage report, so codecov doesn't know about it.

Reply:
Oh interesting, nice catch - I suppose we can adding a coverage report upload from that job as a separate followup task since it's at least running and passing there 😄

Reply:
Opened #8870 to fixup CI coverage reporting.